packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js

```diff
@@ -310,40 +310,50 @@ const applyTransfromHeaders =
   headers =>
     _.mapValues(headers, transformHeaders)
 
-const writeHeadersFile =
-  ({ publicFolder }) =>
-  contents =>
-    new Promise((resolve, reject) => {
-      /**
-       * Emit Headers via IPC
-       */
-      Object.entries(contents).map(([k, val]) => {
-        emitHeaders({
-          url: k,
-          headers: val,
-        })
-      })
+const sendHeadersViaIPC = async headers => {
```

Comment:
GitHub diff here is pretty messy, use `Hide whitespace characters` to make it readable
![image](https://user-images.githubusercontent.com/115628/131387721-b22adf6c-4103-4fa3-9cb7-ce8b3a4681ba.png)